File: 66.plus-one.py
#
# @lc app=leetcode id=66 lang=python
#
# [66] Plus One
#

# @lc code=start
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        strNum = ''
        for i in digits:
            strNum += str(i)
        newStrNum = str(int(strNum) + 1)
        retList = []
        for i in newStrNum:
            retList.append(int(i))
        return retList
        
# @lc code=end

File: 66.plus-one.java
/*
 * @lc app=leetcode id=66 lang=java
 *
 * [66] Plus One
 */

// @lc code=start
class Solution {
    public int[] plusOne(int[] digits) {
        for (int i = digits.length - 1; i>=0;i--) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }
    digits = new int[digits.length + 1];
    digits[0] = 1;
    return digits;
    }
}
// @lc code=end

